### DISCUSSION TITLE
Function failed to boot "failed to read path"

### DISCUSSION THREAD

**Discussion by guillett** [supabase/supabase#29593] [Questions]
Hi,

I had the following bug on one of my edge functions.

`{"code":"BOOT_ERROR","message":"Function failed to start (please check logs)"}`

I managed to solve the problem by rebooting the project (more precisely, I pushed a version with a new function deployed in order to log and debug the original one).

I haven't changed anything.

Here is the error log in the admin console.

`failed to read path`

Here are the error details:

```
{
  "boot_time": null,
  "cpu_time_used": null,
  "deployment_id": "XXXXXXXXXXXXXXX_0edd308c-63c8-4618-b049-913b72575112_6",
  "event_type": "BootFailure",
  "execution_id": "01523883-9ad2-4288-bbbb-708d628fbedf",
  "function_id": "0edd308c-63c8-4618-b049-913b72575112",
  "level": "error",
  "memory_used": [],
  "project_ref": "XXXXXXXXXX",
  "reason": null,
  "region": "eu-west-3",
  "served_by": "supabase-edge-runtime-1.58.4 (compatible with Deno v1.45.2)",
  "timestamp": "2024-09-30T14:50:47.724Z",
  "version": "6"
}
```

How can I know that it won't happen again?
Many thanks!

**Comment by silentworks** [comment]
There is no way to know it won't happen again, but you should open a Supabase support ticket about the issue the next time it happens so that the Supabase team can investigate what could be causing it. https://supabase.com/dashboard/support/new

**Comment by techwizzdom** [comment]
I had the same issue, and realized that the problem was with importing the package, precisely `pdfparse`.

What is interesting is that originally, I had `import pdfparse from "npm:/pdf-parse"`. It worked locally, not when deployed. Total deploy size was ~18MB.

I then changed the import to `import pdfparse from "https://esm.sh/pdf-parse"` and it worked without any issues. Total deploy side reduced to ~9MB.

Since the bundle was so much bigger, **I guess that some garbage was imported that's not compatible with Deno, so it crashed**. It would be nice to have more detailed logs.

I thought that importing with `https://esm.sh/` and `npm:/` is the same, but seems like it is not. Can somebody explain why? Possibly the bundle is so big because other things are imported with `npm:/` too (at least they work).

So @guillett I would recommend:

1. Comment out all the imports,
2. Call the function,
3. It's running and you get a different error? Perfect!
4. Now, add 1 import,
5. Call the function again,
6. Repeat 4-6.

Or just change immediately to `https://esm.sh/` if that's the case for you too.

**EDIT:** it actually doesn't work well. Seems like something is wrong with that package. I just don't know why it works locally, but not when deployed.
